persistence/module.go

```diff
@@ -15,7 +15,10 @@ type persistenceModule struct {
 	bus modules.Bus
 }
 
-func Create(_ *config.Config) (modules.PersistenceModule, error) {
+func Create(c *config.Config) (modules.PersistenceModule, error) {
+	if err := connectAndInitializeDatabase(c.Persistence.PostgresUrl); err != nil {
```

Comment:
To ensure the statelessness of our code, I think we should not include the database "lifecycle" in the module's lifecycle. I'd argue that this will makes things easier especially in development.

In my opinion, we should have all database "scripts" live as separate tools or script, not even necessarily in Go.

For this PR, I would be satisfied with moving this function into a calling script under `tools/database/create.go` and keep everything in one script, so that we separate database and schema creation and validation from the persistence module's lifecycle.

In replacement for the "createIfNoExists" logic wither for the database, schema or tables, I suggest to do basic selects and expect the three to exist, as a validation layer in the `Create` phase of the persistence module, and throw an error indicating that the database is not OK if the database/schema/tables are not created.

About the scripts/tools idea for the database, ideally they should do exactly what this function does, arguably with the possibility of isolating tasks.

- `tools/database/create.(go|sh)`
- `tools/database/migrate.(go|sh)`
- `tools/database/destroy.(go|sh)`

1. Create should do the CreateIfNotExists thing
2. The migration should take care of creating our desired tables, changing their schemas and destroying them.
4. Destroy, wipe out the postgres server clean (_or just the database_)

This is a rough idea, a better idea is to write plain SQL for the schema and migrations and have them be ran optionally at DB start up from docker-compose.

Reply:
I agree - but not for this PR. I'd like to open a follow up issue on this

Reply:
Prior to the schemas being pushed to the same branch & the same PR (different topic), though this suggestion has merit and deserves discussion, was larger than the actual PR reminding me of this: https://twitter.com/iamdevloper/status/397664295875805184?lang=en

I actually asked Andrew to do a small PR so we can just merge in the wiring local for postgres so that we can build on top of it in parallel: e.g. schemas, tooling, migrations, aforementioned design can be done independently while we discuss/review each other's work.

@derrandz For this specific example, I think it's clear that the connection/initialization is not intended to be proper production code in any way, so I think it's safe to close this out because properly designing out the persistence module is a give without needing to be said.